### PR TITLE
Unit Testing with React Testing Library.

### PR DESCRIPTION
**Pull request by mailsg**
Hi,

This PR is raised to demonstrate the implementation of React Testing Library to Unit Test with snapshots with Jest. Below are the pointers implemented :
- Setup React Testing Library
- Written unit test for `operate.js` and `calculate.js`
- Created unit test for React components

Kindly review the PR and provide feedback for the same.

Thanks.

### CODE REVIEW

File: src/logic/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import calculate from './calculate';
```
Comment (gemmen29) [inline src/logic/calculate.test.js:1]: - [ ] Kindly use React Testing Library to simulate user interaction as per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_tests.md#:~:text=Use%20React%20Testing%20Library%20to%20simulate%20user%20interaction.).

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/logic/calculate.test.js (hunk 1 above)
Comment (mailsg) [reply src/logic/calculate.test.js:1]: Done

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @mailsg , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights :fire: 
- All requested changes implemented :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**